--- <CPython-3.4.9>/unittest/test/test_skipping.py
+++ lib/unittest/test/test_skipping.py
@@ -217,6 +217,7 @@
         self.assertEqual(result.unexpectedSuccesses, [test])
         self.assertFalse(result.wasSuccessful())
 
+    @unittest.skip("CircuitPython issue 1179")                                  ###
     def test_skip_doesnt_run_setup(self):
         class Foo(unittest.TestCase):
             wasSetUp = False
@@ -256,5 +257,4 @@
         self.assertEqual(result.skipped, [(test, "testing")])
 
 
-if __name__ == "__main__":
-    unittest.main()
+