--- <CPython-3.4.9>/genericpath.py
+++ lib/genericpath.py
@@ -76,29 +76,16 @@
             return s1[:i]
     return s1
 
-# Are two stat buffers (obtained from stat, fstat or lstat)
-# describing the same file?
-def samestat(s1, s2):
-    """Test whether two stat buffers reference the same file"""
-    return (s1.st_ino == s2.st_ino and
-            s1.st_dev == s2.st_dev)
+# samestat(): Doesn't work because st_ino and st_dev is zero for all files      ###
 
 
 # Are two filenames really pointing to the same file?
 def samefile(f1, f2):
     """Test whether two pathnames reference the same actual file"""
-    s1 = os.stat(f1)
-    s2 = os.stat(f2)
-    return samestat(s1, s2)
+    return (os.path.normcase(os.path.abspath(f1)) ==                            ###
+            os.path.normcase(os.path.abspath(f2)))                              ###
 
 
-# Are two open files really referencing the same file?
-# (Not necessarily the same file descriptor!)
-def sameopenfile(fp1, fp2):
-    """Test whether two open file objects reference the same file"""
-    s1 = os.fstat(fp1)
-    s2 = os.fstat(fp2)
-    return samestat(s1, s2)
 
 
 # Split a path in root and extension.
@@ -130,3 +117,4 @@
             filenameIndex += 1
 
     return p, p[:0]
+