--- <CPython-3.4.9>/test/test_textwrap.py
+++ lib/test/test_textwrap.py
@@ -420,10 +420,12 @@
         self.assertRaises(ValueError, wrap, text, 0)
         self.assertRaises(ValueError, wrap, text, -1)
 
+    @unittest.expectedFailure                                                   ###
     def test_no_split_at_umlaut(self):
         text = "Die Empf\xe4nger-Auswahl"
         self.check_wrap(text, 13, ["Die", "Empf\xe4nger-", "Auswahl"])
 
+    @unittest.expectedFailure                                                   ###
     def test_umlaut_followed_by_dash(self):
         text = "aa \xe4\xe4-\xe4\xe4"
         self.check_wrap(text, 7, ["aa \xe4\xe4-", "\xe4\xe4"])
@@ -930,5 +932,4 @@
         self.check_shorten("Helloo", 5, "[...]")
 
 
-if __name__ == '__main__':
-    unittest.main()
+