--- <CPython-3.4.9>/test/test_dynamic.py
+++ lib/test/test_dynamic.py
@@ -4,7 +4,7 @@
 import contextlib
 import unittest
 
-from test.support import run_unittest, swap_item, swap_attr
+from test.support import swap_item, swap_attr                                   ###
 
 
 class RebindBuiltinsTests(unittest.TestCase):
@@ -135,9 +135,4 @@
         self.assertEqual(foo(), 7)
 
 
-def test_main():
-    run_unittest(RebindBuiltinsTests)
-
-
-if __name__ == "__main__":
-    test_main()
+