--- <CPython-3.4.9>/test/test_dictcomps.py
+++ lib/test/test_dictcomps.py
@@ -1,6 +1,5 @@
 import unittest
 
-from test import support
 
 # For scope testing.
 g = "Global variable"
@@ -74,15 +73,5 @@
         self.assertEqual(actual, expected)
         self.assertEqual(v, "Local variable")
 
-    def test_illegal_assignment(self):
-        with self.assertRaisesRegex(SyntaxError, "can't assign"):
-            compile("{x: y for y, x in ((1, 2), (3, 4))} = 5", "<test>",
-                    "exec")
 
-        with self.assertRaisesRegex(SyntaxError, "can't assign"):
-            compile("{x: y for y, x in ((1, 2), (3, 4))} += 5", "<test>",
-                    "exec")
-
-
-if __name__ == "__main__":
-    unittest.main()
+