--- <CPython-3.4.9>/test/test_unittest.py
+++ lib/test/test_unittest.py
@@ -1,16 +1,15 @@
+import os                                                                       ###
+import unittest                                                                 ###
 import unittest.test
 
 from test import support
 
 
 def test_main():
-    # used by regrtest
-    support.run_unittest(unittest.test.suite())
-    support.reap_children()
+    res = unittest.main(module=None, verbosity=support.verbose, start='/lib/unittest/test', separate=True)  ###
+    errors = len(res.result.errors)                                             ###
+    failures = len(res.result.failures)                                         ###
+    if errors or failures:                                                      ###
+        raise support.TestFailed('Errors {}, Failures {}'.format(errors, failures))  ###
 
-def load_tests(*_):
-    # used by unittest
-    return unittest.test.suite()
-
-if __name__ == "__main__":
-    test_main()
+