--- <CPython-3.4.9>/test/test_int_literal.py
+++ lib/test/test_int_literal.py
@@ -4,7 +4,6 @@
 """
 
 import unittest
-from test import support
 
 class TestHexOctBin(unittest.TestCase):
 
@@ -140,8 +139,4 @@
         self.assertEqual(-0b1000000000000000000000000000000000000000000000000000000000000000, -9223372036854775808)
         self.assertEqual(-0b1111111111111111111111111111111111111111111111111111111111111111, -18446744073709551615)
 
-def test_main():
-    support.run_unittest(TestHexOctBin)
-
-if __name__ == "__main__":
-    test_main()
+