--- <CPython-3.4.9>/json/__init__.py
+++ lib/json/__init__.py
@@ -95,13 +95,6 @@
     $ echo '{ 1.2:3.4}' | python -m json.tool
     Expecting property name enclosed in double quotes: line 1 column 3 (char 2)
 """
-__version__ = '2.0.9'
-__all__ = [
-    'dump', 'dumps', 'load', 'loads',
-    'JSONDecoder', 'JSONEncoder',
-]
-
-__author__ = 'Bob Ippolito <bob@redivi.com>'
 
 from .decoder import JSONDecoder
 from .encoder import JSONEncoder
@@ -329,3 +322,4 @@
     if parse_constant is not None:
         kw['parse_constant'] = parse_constant
     return cls(**kw).decode(s)
+